nevergrad/functions/gym/gym_multi.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import numpy as np
+import os
+import gym
+
+if os.name != "nt":
+    import gym_anm  # pylint: disable=unused-import
+from nevergrad.parametrization import parameter
+from ..base import ExperimentFunction
+
+
+# Method for building a new list, for a future version of gym:
+#
+# GYM_ENV_NAMES = ["gym_anm:ANM6Easy-v0"]
+#
+# for e in gym.envs.registry.all():
+#     try:
+#         assert "Kelly" not in e.id
+#         env = gym.make(e.id)
+#         a1 = env.action_space.sample()
+#         a2 = env.action_space.sample()
+#         a3 = env.action_space.sample()
+#         a1 = a1 + a2 + a3
+#         if hasattr(a1, "size"):
+#             assert a1.size() < 15000
+#         GYM_ENV_NAMES.append(e.id)
+#     except:
+#         pass
+
+GYM_ENV_NAMES = [
+#    "gym_anm:ANM6Easy-v0",   Specific issues for this one!
+    "Copy-v0",
+    "RepeatCopy-v0",
+    "ReversedAddition-v0",
+    "ReversedAddition3-v0",
+    "DuplicatedInput-v0",
+    "Reverse-v0",
+    "CartPole-v0",
+    "CartPole-v1",
+    "MountainCar-v0",
+    "Acrobot-v1",
+    "Blackjack-v0",
+    "FrozenLake-v0",
+    "FrozenLake8x8-v0",
+    "CliffWalking-v0",
+    "NChain-v0",
+    "Roulette-v0",
+    "Taxi-v3",
+    "CubeCrash-v0",
+    "CubeCrashSparse-v0",
+    "CubeCrashScreenBecomesBlack-v0",
+    "MemorizeDigits-v0",
+]
+
+CONTROLLERS = ["conformant", "linear", "neural", "noisy_neural", "noisy_scrambled_neural", "scrambled_neural", "stochastic_conformant"]
+
+
+class GymMulti(ExperimentFunction):
+    def __init__(
+        self, name: str = "gym_anm:ANM6Easy-v0", control: str = "conformant", neural_factor: int = 2
+    ) -> None:
+        env = gym.make(name)
+        self.name = name + "__" + control + "__" + str(neural_factor)
+        self.num_time_steps = 100
+        self.neural_factor = neural_factor
+        o = env.reset()
+        if "int" in str(type(env.action_space.sample())):  # Discrete action space
+            output_dim = env.action_space.n
+            output_shape = (output_dim,)
+            discrete = True
+        else:  # Continuous action space
+            output_shape = tuple(np.asarray(env.action_space.sample()).shape)  # type: ignore
+            discrete = False
+            output_dim = np.prod(output_shape)
+        if "int" in str(type(o)):
```

Comment:
please use the action space

Reply:
It is the observation space and not the action space.
Trying to figure out something.

Reply:
Ok I managed to make it work. But there are environments which do not define the shape, so that I need the original code as a backup.